### DISCUSSION TITLE
Anytone AT-D878UVII Plus steType=off still has a STE tone.

### DISCUSSION THREAD

**Discussion by ikbenkous** [hmatuschek/qdmr#574] [General]
`Anytone AT-D878UVII Plus` still has a STE tone even with `steType` set to `off` as well as with `silent`.

When setting the `steDuration` to `0 ms`, it still defaults to the default value somehow.
I can eliminate the tone with `steFrequency 0`

Is this a firmware issue? I suspect so.
But it could be that qdmr is just not generating the right binary data.

I have no idea and figured I should report this...

**Comment by hmatuschek** [comment]
This is not a bug, but just another issue with the limitations of the extensions. To disable the STE, set the STE type to off. The duration is a value between 10ms and 1000ms. In fact, it is encoded in 10ms steps, starting from 10ms. That is, 0=10ms, 1=20ms, ... So I cannot encode 0ms. 

If you set a value outside of that range, I map it back onto that range. That is, if you set 0ms, I map it to the nearest valid value, i.e. 10ms. This appears to be a bad Idea, but I need to support multiple devices by the same settings. To this end, only the actual device-specific codeplug implementation knows its value range and enforces it. When editing the codeplug in qdmr, qdmr does not know onto which device the codeplug gets written, so it cannot enforce these ranges.

Anyway, I should update the description.

**Comment by ikbenkous** [comment]
Maybe there can be tooltips for each of the settings giving just a generic explanation for all models and how qdmr deals with those values, and if there is a model-specific warning/notice, a little ⚠️ warning or ℹ️ information sign can be drawn? (Doesn't qdmr automatically detect devices?)
